spec/rubocop/cop/style/redundant_return_spec.rb

```diff
@@ -70,45 +98,38 @@ def func
   end
 
   it 'auto-corrects by removing redundant returns' do
-    src = <<~RUBY
+    expect_offense(<<~RUBY)
       def func
         one
         two
         return something
+        ^^^^^^ Redundant `return` detected.
       end
     RUBY
-    result_src = <<~RUBY
+
+    expect_correction(<<~RUBY)
       def func
         one
         two
         something
       end
     RUBY
-    new_source = autocorrect_source(src)
-    expect(new_source).to eq(result_src)
   end
 
   context 'when return has no arguments' do
     shared_examples 'common behavior' do |ret|
-      let(:src) do
-        <<~RUBY
+      it 'registers an offense for #{ret} and autocorrects replacing #{ret} with nil' do
+        expect_offense(<<~RUBY)
           def func
             one
             two
             #{ret}
+            ^^^^^^ Redundant `return` detected.
```

Comment:
Changed to use `%{ret}` but `^{ret}` does not work because in two usages of this shared context - ret = `return` and `return()` both highlight offense on just `return`.

Reply:
🤦‍♂️ ok, sorry :-)